### PR TITLE
Cancel the reconciliation when no running tiller pod in TC

### PR DESCRIPTION
**Pull request by tomahawk28**
Following https://github.com/giantswarm/helmclient/pull/131

### CODE REVIEW

File: service/controller/app/v1/resource/tiller/resource.go [hunk 1]
```diff
@@ -64,6 +64,14 @@ func (r *Resource) ensureTillerInstalled(ctx context.Context, helmClient helmcli
 		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
 		reconciliationcanceledcontext.SetCanceled(ctx)
 
+		return nil
+	} else if helmclient.IsTillerStartingError(err) {
```
Comment (tomahawk28) [inline service/controller/app/v1/resource/tiller/resource.go]: @rossf7 As we discussed in the previous PR, I added in here.

Comment (rossf7) [reply service/controller/app/v1/resource/tiller/resource.go]:
```suggestion
	} else if helmclient.IsTillerNotRunningError(err) {
```

The logic is fine here but the naming I find confusing.

Pods do have a phase but it is running not started. I'd like us to align this with the upstream terminology.

```
kk get po tiller-deploy-584df9b797-n47kw -o yaml | grep phase
  phase: Running
```

File: service/controller/app/v1/resource/tiller/resource.go [hunk 2]
```diff
@@ -64,6 +64,14 @@ func (r *Resource) ensureTillerInstalled(ctx context.Context, helmClient helmcli
...
+		r.logger.LogCtx(ctx, "level", "debug", "message", "no tiller pod in starting phase")
```
Comment (rossf7) [inline service/controller/app/v1/resource/tiller/resource.go]:
```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", "no running tiller pod")
```

File: service/controller/app/v1/resource/tiller/resource.go (hunk 1 above)
Comment (tomahawk28) [reply service/controller/app/v1/resource/tiller/resource.go]: Sure I will change the helmclient library first then.

**Pull request merged**